### PR TITLE
Adjustments to test  data generators

### PR DESCRIPTION
**Pull request by jberthold**
- [X] More diverse patterns from Test.Kore.Internal.Pattern.internalPatternGen
  `Pattern.fromTermLike` would just copy a given term into the `term` part of a `Pattern`. `Pattern.parsePatternFromTermLike` decomposes the top-level `\and`s and splits the term into predicate and "actual term" parts.
- [X] Remove the `#Top` generator from the term generation (as well as the `otherTopSort` sort from the generator setup)
- [x] Ensure `Test.ConsistentKore.patternGen` does not loop and is able to generate requested sizes.
- [X] Avoid using term connectives (`\and`, `\or`) inside a function (holds for both map- and set-constructing symbols as well as built-in operations like `<Int`)

This can be merged as-is , and fixes part of #3259

---

###### Review checklist

The author performs the actions on the checklist. The reviewer evaluates the work and checks the boxes as they are completed.

-   [ ] **Summary.** Write a summary of the changes. Explain what you did to fix the issue, and why you did it. Present the changes in a logical order. Instead of writing a summary in the pull request, you may push a clean Git history.
-   [ ] **Documentation.** Write documentation for new functions. Update documentation for functions that changed, or complete documentation where it is missing.
-   [ ] **Tests.** Write unit tests for every change. Write the unit tests that were missing before the changes. Include any examples from the reported issue as integration tests.
-   [ ] **Clean up.** The changes are already clean. Clean up anything near the changes that you noticed while working. This does not mean only spatially near the changes, but logically near: any code that interacts with the changes!

### CODE REVIEW

File: kore/test/Test/Kore/Rewrite/MockSymbols.hs [hunk 1]
```diff
@@ -2325,7 +2331,7 @@ generatorSetup =
     ConsistentKore.Setup
         { allSymbols = filter doesNotHaveArguments symbols
         , allAliases = []
-        , allSorts = map fst sortAttributesMapping
+        , allSorts = filter (/= otherTopSort) $ map fst sortAttributesMapping
```
Comment (jberthold) [inline kore/test/Test/Kore/Rewrite/MockSymbols.hs:2334]: Note that we still have this `otherTopSort` around in the tests. 
https://github.com/runtimeverification/haskell-backend/blob/ed3edb3e59eafbe0992b8d8aec458cb79773067b/kore/test/Test/Kore/Rewrite/MockSymbols.hs#L2112..L2128 appears a bit strange to me, not well-ordered at least.

Comment (goodlyrottenapple) [reply kore/test/Test/Kore/Rewrite/MockSymbols.hs:2334]: I like `overTheTopSort` :D

File: kore/test/Test/ConsistentKore.hs [hunk 2]
```diff
@@ -818,14 +794,18 @@ symbolGenerator
...
+                    | BuiltinMap.isSymbolElement symbol =
+                        requestConcrete . requestConstructorLike
+                    | BuiltinSet.isSymbolElement symbol =
+                        requestConcrete . requestConstructorLike
+                    | isFunction =
+                        withoutConnectives
+                    | otherwise = id
+                -- TODO (virgil): also allow constructor-like
+                -- stuff with variables.
+                isFunction = Attribute.Symbol.isFunction symbolAttributes
```
Comment (goodlyrottenapple) [inline kore/test/Test/ConsistentKore.hs:807]: why not inline above?

Comment (jberthold) [reply kore/test/Test/ConsistentKore.hs:807]: Good point... the separation was from before when I hadn't found that `isFunction` is ready-made. Will fix it.

**Review by ana-pantilie — APPROVED** [review]

**Pull request merged**